net/netclient/Makefile

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 Stan Grishin (stangri@melmac.ca)
+# This is free software, licensed under the Apache-2.0 License.
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=netclient
+PKG_VERSION:=0.22.0
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/gravitl/netclient/tar.gz/v$(PKG_VERSION)?
+PKG_HASH:=010e7ee72f8cf8ae4cc91b19010b85875d75e70faabd248c992af2487629211d
+
+PKG_MAINTAINER:=Stan Grishin <stangri@melmac.ca>
+PKG_LICENSE:=Apache-2.0
+PKG_LICENSE_FILES:=LICENSE.txt
+
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+PKG_BUILD_FLAGS:=no-mips16
+
+GO_PKG:=github.com/gravitl/netclient
+GO_PKG_BUILD_PKG:=github.com/gravitl/netclient
+GO_PKG_LDFLAGS_X:=main.version=v$(PKG_VERSION)
+
+include $(INCLUDE_DIR)/package.mk
+include ../../lang/golang/golang-package.mk
+
+define Package/netclient
+  SECTION:=net
+  CATEGORY:=Network
+  TITLE:=netclient
+  URL:=https://docs.openwrt.melmac.net/netclient/
+  DEPENDS:=$(GO_ARCH_DEPENDS) +wireguard-tools
+endef
+
+define Package/netclient/description
+This is the client for Netmaker networks. To learn more about Netmaker, see URL below:
+https://github.com/gravitl/netmaker
+endef
+
+define Package/netclient/install
+	$(call GoPackage/Package/Install/Bin,$(PKG_INSTALL_DIR))
+	$(INSTALL_DIR) $(1)/etc/init.d
+	$(INSTALL_BIN) ./files/netclient.init $(1)/etc/init.d/netclient
+	$(SED) "s|^\(readonly PKG_VERSION\).*|\1='$(PKG_VERSION)-$(PKG_RELEASE)'|" $(1)/etc/init.d/netclient
```

Comment:
Wonder about the sed, which is why I already asked.

Reply:
The sed is there so I wouldn't have to manually modify the version in the init script.

Reply:
Wouldnt it be better to rather specify the version in the init script instead of doing sed? Like it is done for syslog-ng config file? But still, you should not handle the version at all here or in the init script.

Reply:
My concern is that I'll forget to edit the version in the init script. The version number is needed there because the binary needs to present its version in a way acceptable by server and it fails the CI test script then, so I need a version number in the init script. The sed worked for other packages. Is there a compelling reason not to do that?

Reply:
Just a drive-by comment here, but won't the git version suffice? Or just build date?